### DISCUSSION TITLE
Cropping to faces

### DISCUSSION THREAD

**Discussion by brendon** [janko/image_processing#119] [Q&A]
I currently have a paperclip processor called `face_cropper`. It makes a local call to a little express app that I wrote that uses `@vladmandic/human` to find the primary face in the image and then sets the centre of the crop to that face. It works well.

I want to do the same thing with Active Storage, for example:

`image.variant(resize_to_fill: [932, 466, crop: :face])`

I'm using vips currently and wondered if `image_processing` would be the correct layer to attempt to add this processor?

**Comment by janko** [comment]
The `image_processing` pipeline relies on building a single ImageMagick/Vips command with the given operations. In your case, you'd need to call your local processor and `image_processing` separately. I'm not sure you'd customize that with Active Storage.

**Comment by brendon** [comment]
Thanks @janko. I'm thinking down the path of an extension to the `ImageAnalyzer` that adds some metadata on where the primary face is. This can then be used to inform the crop call I suspect. (hopefully!)

**Comment by brendon** [comment]
I've managed to make some good progress and have extended the Active Storage `ImageAnalyzer` to also find the focal point which is just an array of two 0..1 floats indicating the X and Y location of the most prominent face. I'd like to convert this into the css `object-position` as percentages along with an image that is resized to cover a minimum pixel area. Unfortunately it seems that `resize_to_fill` with `crop: :none` resizes the image down so that it's contained within those two dimensions rather than creating an image that covers the two dimensions. Is that by design, and if so, would you consider a PR that adds a `resize_to_cover` method? :)

**Reply by brendon** [reply]
I've gone ahead and had a first attempt at this. It works, but the tests are where I'm stuck:

https://github.com/janko/image_processing/pull/120
